### DISCUSSION TITLE
Solving Te in a second order differential equation

### DISCUSSION THREAD

**Discussion by youareit125** [idaholab/moose#25674] [Q&A General]
I have been working on creating a plasma model and I am stuck solving Te(t). I was wondering if I could get some help implementing an equation below.

![Diff_Eq](https://github.com/idaholab/moose/assets/86081352/2504538f-b958-4a2a-b12e-fd7724b21197)
> e, Ar, Ar+ are variables from the input file below and Tg = 500.

I would like to solve for Te(t) using the equation above, then plug Te(t) into rate coefficient equations in the reaction section. Is there a way to use the existing MOOSE kernels to solve this?

```
[Mesh]
  type = GeneratedMesh
  dim = 1
  xmin = 0
  xmax = 1
  nx = 1
[]

[Variables]
  [Ar+]
    family = SCALAR
    order = FIRST
    initial_condition = 1e6
  []
  [Ar2+]
    family = SCALAR
    order = FIRST
    initial_condition = 1e13
  []
  [Ar]
    family = SCALAR
    order = FIRST
    initial_condition = 9.05e+20
  []
[]

[ScalarKernels]
  [dAr+_dt]
    type = ODETimeDerivative
    variable = Ar+
  []
  [dAr2+_dt]
    type = ODETimeDerivative
    variable = Ar2+
  []
  [dAr_dt]
    type = ODETimeDerivative
    variable = Ar
  []
[]

[ChemicalReactions]
  [ScalarNetwork]
    species = 'Ar+ Ar2+ Ar'
    aux_species = 'e'
    interpolation_type = 'spline'
    # These are parameters required equation-based rate coefficients
    equation_constants = 'Tg'
    equation_values = '500'
    equation_variables = 'Te'
    reactions = 'e + Ar -> e + e + Ar+      :{2.34e-14*Te^(0.59)*exp(-17.44/Te)}
    e + e + Ar+ -> e + Ar      :{2e-39*Te^(-4.5)}
    e + Ar + Ar+ -> Ar + Ar    :{1.5e-40*(Tg/300)^(-2.5)}
    Ar+ -> Ar                  :{1/(1.95e7*Te + 8.41e5)}'
  []
[]

[AuxVariables]
  [e]
    order = FIRST
    family = SCALAR
  []
[]

[AuxScalarKernels]
  [Quasi_neutrality]
    type = VariableSum
    variable = e
    args = 'Ar+ Ar2+'
    execute_on = 'INITIAL TIMESTEP_END'
  []
[]

[Executioner]
  type = Transient
  automatic_scaling = true
  end_time = 0.001
  solve_type = 'newton'
  dtmin = 1.0e-20
  dtmax = 1e-06
  petsc_options_iname = '-snes_linesearch_type'
  petsc_options_value = 'basic'
[]

[Preconditioning]
  [smp]
    type = SMP
    full = true
  []
[]

[Outputs]
  csv = true
[]
```

**Comment by GiudGiud** [comment]
Hello

This will take some coding work to get all the scalar kernels in that equation. 
The first term is this scalar kernel:
https://mooseframework.inl.gov/source/scalarkernels/ADScalarTimeDerivative.html
The second term will need to be coded, and you ll want to use this routine `coupledScalarDotDot` from the scalarCoupleable interface

The 3rd 4th and 5th term all have a functional form, which while simple has not been written down in an existing kernel. 
So you ll need to make a few scalarkernels for these too.

But overall, this is very reasonable to do in MOOSE. If you followed the tutorial you know how to make regular kernels, you ll just want to adapt that knowledge to make scalar kernels instead (different base class, slightly different routine names)

Guillaume
